### ISSUE TITLE
Building for arm64 device, arm64 simulator and x86_64 simulator

### ISSUE THREAD

**Issue by mungler** [google/j2objc#1830]
Following on from https://github.com/google/j2objc/issues/1828 ...

I'm not sure I was exactly clear, or @tomball maybe read in a hurry :) My problem isn't that I can't build for arm64 on M1. My problem is that building on an Intel Mac ONLY builds for arm64, for some reason.

For completeness, I followed your suggestion in the last ticket and executed:

```J2OBJC_ARCHS="iphone64 simulator64 macosx64 maccatalyst64" make -j8 frameworks```

And the resulting built frameworks appear to have arm64 binaries only:

```
rory@Eddie ~/W/j2objc (master)> ls dist/frameworks/JRE.xcframework/
Headers			Versions		ios-arm64-simulator
Info.plist		ios-arm64
```

This is weird, no? Why on earth am I only getting arm64 binaries for device and simulator? This also happens on my M1 machine, for reference.

Thanks!

***

Original ticket copy below:

We're in the midst of trying to update our j2objc library to XCFramework, and are hoping to be able to run our app natively on the arm64 Simulator on M1 Macs. I'm trying to build j2objc from source (latest master as of today - commit f119c4274dde21785dc8d96bef52f3dc3a15ecac).

I ran the following:

```
J2OBJC_ARCHS="iphone64 simulator64" make -j8 frameworks
```

However, no matter if i build on my M1 Air or my Intel iMac, this results in XCFrameworks with only arm64 binaries:

```
rory@Eddie ~/W/j/d/frameworks (master)> ls JRE.xcframework/
Headers			Versions		ios-arm64-simulator
Info.plist		ios-arm64
rory@Eddie ~/W/j/d/frameworks (master)> file JRE.xcframework/ios-arm64-simulator/libjre_emul.a 
JRE.xcframework/ios-arm64-simulator/libjre_emul.a: Mach-O universal binary with 1 architecture: [arm64:current ar archive]
JRE.xcframework/ios-arm64-simulator/libjre_emul.a (for architecture arm64):	current ar archive
rory@Eddie ~/W/j/d/frameworks (master)> file JRE.xcframework/ios-arm64/libjre_emul.a 
JRE.xcframework/ios-arm64/libjre_emul.a: Mach-O universal binary with 1 architecture: [arm64:current ar archive]
JRE.xcframework/ios-arm64/libjre_emul.a (for architecture arm64):	current ar archive
rory@Eddie ~/W/j/d/frameworks (master)> 
```

I first did the build on my M1 so I thought it was just picking up on the local CPU arch, but I repeated on my intel iMac and got the same result.

Our own library build (including j2objc-converted Java code) generates an XCFramework with a fat-binary for Simulator:

```
rory@Hactar ~/W/l/b/LibASW-iOS.xcframework> ls
Info.plist			ios-arm64_x86_64-simulator
ios-arm64
... etc
```

though admittedly this is largely built 'by hand' in our build script.

Am I doing something wrong? My goal is to use JRE.xcframework in our app, along with our own library, and have it run natively on both Intel and Apple Silicon Macs.

**Comment by mungler** [comment]
Giving it a try with defaults, i.e. no explicit J2OBJC_ARCHS env var set... will report results shortly.

**Comment by mungler** [comment]
Ok, that took a long time, and eventually failed with:

```building jre_emul archs: macosx iphone64 iphone64e watchosv7k watchos64 watchsimulator watchsimulator64 simulator simulator64 maccatalyst appletvos appletvsimulator macosx64 maccatalyst64
building JRE framework
xcodebuild -create-xcframework -output ../dist/frameworks/JRE.xcframework -library /Users/rory/Workspace/j2objc/jre_emul/build_result/iphone/libjre_emul.a -library /Users/rory/Workspace/j2objc/jre_emul/build_result/simulator/libjre_emul.a -library /Users/rory/Workspace/j2objc/jre_emul/build_result/maccatalyst/libjre_emul.a -library /Users/rory/Workspace/j2objc/jre_emul/build_result/watchos/libjre_emul.a -library /Users/rory/Workspace/j2objc/jre_emul/build_result/watchsimulator/libjre_emul.a -library /Users/rory/Workspace/j2objc/jre_emul/build_result/appletvos/libjre_emul.a
error: binaries with multiple platforms are not supported '/Users/rory/Workspace/j2objc/jre_emul/build_result/appletvos/libjre_emul.a'
make[1]: *** [../dist/frameworks/JRE.xcframework] Error 70
make: *** [frameworks] Error 2
```

**Comment by mungler** [comment]
Ok, I just realised maybe im misunderstanding what `simulator64` actually means. I thought it meant build for 64-bit platforms using the Simulator SDK. But does it actually exclusively mean build for arm64 with Simulator SDK?

Going to try a build with

```
J2OBJC_ARCHS="simulator simulator64 iphone64"
```

**Comment by mungler** [comment]
Yep! Problem solved, sorry for the misunderstanding.

```rory@Eddie ~/W/j2objc (master)> ls dist/frameworks/JRE.xcframework/
Headers				Versions			ios-arm64_x86_64-simulator
Info.plist			ios-arm64
```

Will close now.

**Comment by tomball** [comment]
FYI: the failure building appletvos should be fixed in PR #1829, now being submitted.

**Comment by mungler** [comment]
Good to know, thanks @tomball
